packages/webview_flutter/webview_flutter_wkwebview/lib/src/web_kit_webview_widget.dart

```diff
@@ -120,6 +121,36 @@ class WebKitWebViewPlatformController extends WebViewPlatformController {
   @visibleForTesting
   late final WKUIDelegate uiDelegate = webViewProxy.createUIDelgate();
 
+  /// Methods for handling navigation changes and tracking navigation requests.
+  @visibleForTesting
+  late final WKNavigationDelegate navigationDelegate =
+      webViewProxy.createNavigationDelegate()
+        ..didStartProvisionalNavigation = (WKWebView webView) {
+          webView.url.then<void>(
+            (String? url) => callbacksHandler.onPageStarted(url ?? ''),
```

Comment:
An alternative to needing to make another method call to retrieve the url would be to create a custom method in `WKNavigationDelegate` that also includes the url.

Reply:
I think this is a case where we are going to actually need to flex the boundary away from the direct wrapping slightly, because of the async calls. This won't actually give correct results in all cases, because If there's a sequence of redirects there's no guarantee that the URL that is retrieved asynchronously is actually the one that the delegate callback was actually about.

Reply:
I agree. I updated the method